crates/foundry-ui/src/components/tagged.rs

```diff
@@ -0,0 +1,40 @@
+use console::Style;
+use serde::Serialize;
+
+use crate::Message;
+
+/// Generic textual message with `tag` prefix.
+///
+/// The type prefix can be stylized in text mode.
+/// e.g. "[WARNING]: An example warning message"
+#[derive(Serialize)]
+pub struct TaggedMessage<'a> {
+    tag: &'a str,
+    text: &'a str,
+    #[serde(skip)]
+    type_style: Option<&'a str>,
+}
+
+impl<'a> TaggedMessage<'a> {
+    #[must_use]
+    pub fn styled(tag: &'a str, text: &'a str, type_style: &'a str) -> Self {
```

Comment:
This is cool but if we want to have consistent styling we would probably want to have some enum with predefined styles fir different things in the future.

Reply:
To me it makes more sense when these styles are predefined inside specific print methods on `UI` (e.g. `print_warning`, `print_error`), wdyt?

Reply:
Yeah it probably doesn't make sense now. But if we eventually get a lot of styles then it will make sense to abstract that somehow probably.

Reply:
Created https://github.com/foundry-rs/starknet-foundry/issues/3408